### ISSUE TITLE
Support for ARM 32 bit (armhf ) 

### ISSUE THREAD

**Issue by aslanpour** [kedacore/keda#3334]
### Proposal

Hi All, 

Thank you for the amazing job you have done to develop KEDA.
I just tried to install KEDA on my Raspberry Pi cluster that uses Kubernetes K3s, but apparently KEDA docker images are not built for ARM 32-bit architecture (it already supports ARM 64-bit). I would appreciate it if you would consider this support.

### Use-Case

I would like to run KEDA on devices with ARM 32-bit architecture.

**Comment by JorTurFer** [comment]
I have checked the supported arch of `gcr.io/distroless/static:nonroot` and also supports `arm` so I'd say that it's doable. My only concern is that we don't have any machine with that arch to run at least the smoke test... 
WDYT @kedacore/keda-core-contributors ? 
I have tried building it from my local and at least it can be built only adding the new target

**Comment by zroubalik** [comment]
We can try to ask CNCF if they have some spare machine to cover this?

**Comment by JorTurFer** [comment]
> We can try to ask CNCF if they have some spare machine to cover this?

Does it make sense? I mean, do we need to run smoke test as a requirement for this arch? AFAIK, it's not an arch to use in prod. I agree that test everything is always better, but if it's not doable I'm not sure if it's a blocker in this case

**Comment by tomkerkhove** [comment]
> > We can try to ask CNCF if they have some spare machine to cover this?
> 
> Does it make sense? I mean, do we need to run smoke test as a requirement for this arch?

Yes, everything that we offer should be verified and tested IMO.

> AFAIK, it's not an arch to use in prod. I agree that test everything is always better, but if it's not doable I'm not sure if it's a blocker in this case. But maybe I'm wrong and it's used in prod, in any case, we could add a disclaimer for it

Who says this will not be used in production? 😄 We cannot make that assumption if we offer it as part of our image imo.

**Comment by zroubalik** [comment]
> Who says this will not be used in production? 😄 We cannot make that assumption if we offer it as part of our image imo.

💯

**Comment by JorTurFer** [comment]
let's open a ticket to CNCF asking for the machine then 😄

**Comment by tomkerkhove** [comment]
Did you check if Equinix has ARM 32 machines already?

**Comment by JorTurFer** [comment]
> Did you check if Equinix has ARM 32 machines already?

I have checked and I think that they don't offer it https://metal.equinix.com/product/servers/

**Comment by tomkerkhove** [comment]
Ok, feel free to open a ticket with CNCF in that case.

**Comment by JorTurFer** [comment]
I have reviewed the previous request for ARM64 and checking the catalogue there, there isn't any ARM32 machine in Equinix. I'm going to open a support ticket

**Comment by JorTurFer** [comment]
They answered me redirecting me to the CNCF repository and there they request the server model from Equinix catalogue, but Equinix doesn't have arm32 machines. I'll check if we can use emulation for running this smoke tests but checking, kind doesn't have either an image for arm32, we should compile it if we want use.
I think that we could give another try in few months, but atm it could be a lot of work

**Comment by jeffhollan** [comment]
To confirm as well, newer Raspberry Pis support ARM64 correct? Not saying we shouldn't do this if we can, but wondering if this is a point-in-time limitations and hopefully newer gens will be good with current ARM bits

**Comment by JorTurFer** [comment]
wow, you're right @jeffhollan , this is locked indeed due to hardware limitations that we have to test them. Maybe we could emulate ARM32 in the ARM machine to support it, but for sure this shouldn't be high prio
